avocado/utils/software_manager.py

```diff
@@ -658,6 +743,63 @@ def provides(self, name):
         except process.CmdError:
             return None
 
+    def _source_install(self, name):
+        """
+        Source install the given package [name]
+        Returns the SPEC file of the package
+
+        :param name: name of the package
+
+        :return path: path of the spec file
+        """
+        s_cmd = '%s source-install %s' % (self.base_command, name)
+
+        try:
+            process.system(s_cmd, sudo=True)
+            s_cmd = '%s source-install -d %s' % (self.base_command, name)
+            process.system(s_cmd, sudo=True)
+            return '/usr/src/packages/SPECS/%s.spec' % name
+        except process.CmdError:
+            log.error('Installing source failed')
+            return ""
+
+    def prepare_source(self, name, dest_path):
```

Comment:
Both the `_source_install` and the `prepare_source` can and should share most of the code.

Reply:
Can you elaborate here? Not sure about commonality. Thanks.

Reply:
This was meant for the previous version

Reply:
Actually not, it's this version.

Reply:
What I had in mind was to create generic `prepare_source` in `RpmBackend` and only call the specific functions from each backends. The whole `rpmbuild -bp ...` is shared among `Yum`, `Dnf` and `Zypper`.

Reply:
(and the `check_installed` can be called from the inherited classes before calling the super)

Reply:
@ldoktor Okay, makes sense. I will work on this. Thanks.